### PR TITLE
[ADD] l10n_be_mis_reports_xml: new module

### PR DESCRIPTION
**Pull request by robinkeunen**
This module extends `l10n_be_mis_reports` to generate xml files that can be uploaded to Belgium official platforms. This PR only implements periodical VAT declaration based on `mis_report_vat`.

If the mis.report.instance's report_id is `mis_report_vat`, a button is visible. It calls `be.vat.declaration.wizard` to collect required values. The generated xml looks like this (demo data):

```xml
<?xml version='1.0' encoding='UTF-8'?>
<VATConsignment xmlns="http://www.minfin.fgov.be/InputCommon" VATDeclarationsNbr="1">
    <VATDeclaration SequenceNumber="1" DeclarantReference="20210303000010">
        <Declarant>
            <Name>YourCompany</Name>
            <Street>1725 Slough Ave.</Street>
            <PostCode>18540</PostCode>
            <City>Scranton</City>
            <CountryCode>US</CountryCode>
            <Phone>+1 555 123 8069</Phone>
        </Declarant>
        <Period>
            <Quarter>4</Quarter>
            <Year>2020</Year>
        </Period>
        <Data>
            <Amount GridNumber="59">867.77</Amount>
            <Amount GridNumber="72">867.77</Amount>
            <Amount GridNumber="82">4132.22</Amount>
            <Amount GridNumber="91">4.3</Amount>
        </Data>
        <ClientListingNihil>NO</ClientListingNihil>
        <Ask Restitution="NO" Payment="NO"/>
    </VATDeclaration>
</VATConsignment>
```

The xml schemas are [documented here](https://finances.belgium.be/fr/E-services/Intervat/documentation-technique). As the xsd is only relevant for Belgium, this module has its place in l10n-belgium (rather than mis-builder).

I put the code in a separate module to keep `l10n_be_mis_reports` because it adds a few wizards and fields.

Comments are welcome.

### CODE REVIEW

File: l10n_be_mis_reports_xml/wizards/be_vat_declaration_wizard.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# Copyright 2021 Coop IT Easy SCRL fs
+#   Robin Keunen <robin@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import api, fields, models
+
+
+class BeVATDeclarationWizard(models.Model):
```
Comment (sbidoul) [inline l10n_be_mis_reports_xml/wizards/be_vat_declaration_wizard.py]: Should this not be a transient model ?

**Review by sbidoul — APPROVED** [review]
Nice! I did a quick review and this looks good except the wizard declaration.

**Comment by sbidoul** [comment]
@robinkeunen I realize I'm alone in the l10n-belgium PSC. @robinkeunen would you like to join it ? And maybe @vdewulf would like to be PSC representative ?

**Comment by robinkeunen** [comment]
Hi @sbidoul, it'll be my first time in a PSC, I had a look at the [PSC guide](https://odoo-community.org/page/oca-project-steering-committee-guide). If I understand correctly, the role is to review the PR and oversee the consistency of the developments in this repository?

If so, thank you for the invitation, I can join the PSC. :-)

The wizard model is fixed, good catch.

**Review by sbidoul — APPROVED** [review]

**Comment by sbidoul** [comment]
@robinkeunen great! I configured your user in the [OCA database](https://odoo-community.org/psc-teams/belgium-50). It should sync to GitHub by tomorrow.

**Comment by houssine78** [comment]
@robinkeunen I did a quick review LGTM. My only question is that you declare a xsd file but I didn't saw any validation of the generated xml using that xsd.

**Review by houssine78 — APPROVED** [review]

**Comment by robinkeunen** [comment]
@houssine78 Indeed, I reference the xsd of the belgian administration. It would be a good improvement to validate the XML. I'll add it to the roadmap.

I'll check the runbot warnings next week.

**Comment by vdewulf** [comment]
The generated XML has been tested and imported in Intervat platform by our accountant without any issue, thanks to the last changes made:
- Field <VATNumber> in the <ns2:Declarant> zone: it takes only the numbers from the company settings, and ignore the other caracters.
- Field <Phone> in the <ns2:Declarant> zone: it takes only the numbers from the company settings, and ignore the other caracters.

The only important point is about the status of the Declarant. When the declaration is made by a "mandataire" (in french), so for example the accountant), this person will have to choose the option "Recharger dans les écrans", confirm the email adress of the declarant, the email adress of the representative and his phone number.
If this option is not choosen, there will be an error like " (" Vous êtes dans le contexte "Mandataire", les données concernant le mandataire doivent être remplies.")

We still will have to test the intervat process when a legal representative of the company does the declaration himself, and if the same flow is to be taken or not.
We will have this information when our customers use the module in production.

For me, the module is validated as is and might be improved later on.

@robinkeunen Could your proceed for the merge as PSC please if we're ready to go?
Thanks

**Comment by houssine78** [comment]
@vdewulf @robinkeunen runbot should be first fixed. I'm not sure that a psc member could merge its own PR though.

**Comment by robinkeunen** [comment]
Hi @sbidoul , it looks like I don't have the the rights to merge even though I'm in the [PSC](https://odoo-community.org/psc-teams/belgium-50). Do you know what's going on? Should I add the maintainer key to this module?

**Comment by sbidoul** [comment]
I'm looking into it. You had duplicate contacts in the OCA database. Should should be in the team now.

**Comment by robinkeunen** [comment]
Thanks, I got the notification: it's merging now :-)

**Pull request merged**